packages/fxa-settings/src/pages/InlineTotpSetup/index.tsx

```diff
@@ -0,0 +1,397 @@
+   *  - add in actions for `Cancel` `Ready` etc
+   *  - fetch code here if that's preferable
+   */
+
+  const ftlMsgResolver = useFtlMsgResolver();
+  const localizedSecurityCodePlaceholder = ftlMsgResolver.getMsg(
+    'inline-totp-setup-security-code-placeholder',
+    'Security code'
+  );
+  const localizedQRCodeAltText = ftlMsgResolver.getMsg(
+    'inline-totp-setup-qr-code-alt-text',
+    `Use the code ${code} to set up two-step authentication in supported applications.`,
+    { code }
+  );
+  const localizedTwoFactorImgAriaLabel = ftlMsgResolver.getMsg(
+    'inline-totp-setup-image-label',
+    'A device with a hidden 6-digit code.'
+  );
+  const [secret, setSecret] = useState<string>();
+  const [qrCodeSrc, setQRCodeSrc] = useState<string>();
+  const [showIntro, setShowIntro] = useState<boolean>(true);
+  const [totpCodeValue, setTotpCodeValue] = useState<string>('');
+  const [showQR, setShowQR] = useState<boolean>(true);
+  const [totpErrorMessage, setTotpErrorMessage] = useState<string>('');
+
+  const [isFocused, setIsFocused] = useState(false);
+
+  const onFocusMetricsEvent = 'inline-totp-setup-code.engage';
+
+  const onFocus = () => {
+    if (!isFocused && onFocusMetricsEvent) {
+      logViewEvent('flow', onFocusMetricsEvent, {
+        entrypoint_variation: 'react',
+      });
+      setIsFocused(true);
+    }
+  };
+
+  type FormData = {
+    confirmationCode: string;
+  };
+
+  const { handleSubmit } = useForm<FormData>({
+    mode: 'onBlur',
+    criteriaMode: 'all',
+    defaultValues: {
+      confirmationCode: '',
+    },
+  });
+
+  const onSubmit = () => {
+    if (!totpCodeValue) {
+      // TODO: Add l10n for this string
+      // Holding on l10n pending product decision
+      // See FXA-6422, and discussion on PR-14744
+      setTotpErrorMessage('Backup authentication code required');
+    }
+    try {
+      // Check security code
+      // logViewEvent('flow', inline-totp-setup.submit, {
+      //   entrypoint_variation: 'react',
+      //  });
+    } catch (e) {
+      // TODO: error handling, error message confirmation
+      //       - use the Banner component
+      // const errorInlineTotpSetup = ftlMsgResolver.getMsg(
+      //   'inline-totp-setup-error-general',
+      //   'Invalid confirmation code'
+      // );
+    }
+  };
+
+  return (
+    <>
+      {showIntro && (
+        <>
+          {serviceName ? (
+            <FtlMsg
+              id="inline-totp-setup-enable-two-step-authentication-custom-header"
+              vars={{ serviceName }}
+              elems={{
+                enable2StepCustomServiceSpan: (
+                  <span className="card-subheader">{`to continue to ${serviceName}`}</span>
+                ),
+              }}
+            >
+              <h1 className="card-header">
+                {/* <!-- L10N: For languages structured like English, the second phrase can read "to continue to %(serviceName)s" --> */}
```

Comment:
nit: Is this comment accurate? Or copy-pasta from some .po files? Looks like the old `%(serviceName)s` syntax.

Reply:
These comments used to be extracted into our .po files for l10n but they won't see any in our TSX anymore, they have to be in the FTL files.

I think this can all be replaced by the `CardHeader` component.

Reply:
Replaced!